Dockerfile

```diff
@@ -13,6 +13,22 @@ ENV ODK_VERSION $ODK_VERSION
 # docker run -v $HOME/.coursier/cache/v1:/tools/.coursier-cache ...
 ENV COURSIER_CACHE "/tools/.coursier-cache"
 
+# nvm environment variables
+ENV NVM_DIR /usr/local/nvm
+ENV NODE_VERSION 16.17.0
+
+# install nvm, node and npm
+RUN mkdir $NVM_DIR \
+    && curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.1/install.sh | bash \
```

Comment:
Running an arbitrary piece of code just downloaded from the Internet at every build is a no-go. Feel free to do that on your own machine if you don’t care, but not when building an image that will end up on the machine of every ODK user.

Just because everything runs within a container does not mean we are automatically immune to malware. Keep in mind that normal usage of the ODK implies binding a part of the host’s filesystem into the container, so the user’s files are _not_ protected from any malware that could have crept into the ODK.

Supply-chain attacks are trendy, and while we cannot eliminate them altogether (we depend on so many programs and packages that there will always be a risk), let’s at least avoid expanding the attack surface.

Reply:
Thank you so much of the explanation. That makes total sense.

Reply:
No! You’re still executing a script straight from the Internet.

What you should do here is to download this script (_without executing it!_), examine it to understand what it is supposed to do (here, it adds a new APT source), and then write your own code in the Dockerfile to do the same thing, without ever needing that script again.

Reply:
See commit 1fc9a8ac0e2e880029b779af6912b5b285a8381a (that I just added) for the proper way of doing it (there was an additional difficulty due to the need of handling Nodesource’s signing key).

Reply:
Thanks! I'm clearly new to this so I sincerely appreciate this!

Reply:
No problem! Sorry for being quite a pain in the neck about this issue. :) Your method was working, it was just not following good practices, security-wise.

Reply:
No problem at all! As I mentioned earlier, I truly appreciate it. So do you think this PR is ready to merge?

Reply:
I manually triggered a run of the CI test suite, just to be sure. (https://github.com/INCATools/ontology-development-kit/actions/runs/3055880201) (it should have been triggered automatically by GitHub after I added the last commit, but for some reasons it has not). Once the run completes (and assuming it passes, obviously), yes, it will be ready to merge.

Reply:
All clear. You can go ahead and merge. :tada:

Reply:
THANK YOU BOTH!